### PR TITLE
docs: adjust 2.x docs for 3.x

### PR DESCRIPTION
**Pull request by kacperkapusciak**
## Summary

This PR slightly adjusts Reanimated 2.x documentation to be more appropriate for the Reanimated 3.x

This is mostly done by changing use of "Reanimated 2" to just "Reanimated". This PR also:
- updates the About page 
- adds a short `migration from 2.x` page
- move migration pages to Guides

### CODE REVIEW

File: docs/docs/fundamentals/about.md [hunk 1]
```diff
@@ -6,15 +6,15 @@ sidebar_label: About
...
 
-If you are looking for Reanimated 1 docs [please follow this link](https://docs.swmansion.com/react-native-reanimated/docs/1.x/).
+Looking for an older version? Check the [Reanimated 2.x](/docs/2.x/) or [Reanimated 1.x](/docs/1.x/).
 
 :::
 
 :::caution
 
-Reanimated v2 only supports `react-native` `0.62+`
+Reanimated 3.x aims to support [3 last stable React Native versions](https://github.com/reactwg/react-native-releases#which-versions-are-currently-supported).
```
Comment (tomekzaw) [inline docs/docs/fundamentals/about.md:17]: Reanimated 3.0.0 aims to support RN 0.64+ but we don't make such promise for the next releases.

File: docs/docs/fundamentals/about.md [hunk 2]
```diff
@@ -33,40 +33,18 @@ Reanimated aims to provide ways of offloading animation and event handling logic
...
-   The style worklet will update whenever shared values used by that worklet change (we detect dependencies on shared values automatically).
-4. Animations can be started in two ways: by triggering animated change on a shared value, or by returning animated value from `useAnimatedStyle` hook.
-5. With reanimated, we can hook worklets to serve as event handlers.
-   Most common case for an event worklet is to modify some shared values.
-   As a result, changes made to those values will be reflected in the animated style worklet being triggered, which in turn will result in some view properties being updated.
-   For convenience, Reanimated provides an event hook that is tailored to work together with Gesture Handler library and allows you to define a separate worklet for handling different handler states (e.g., onStart, onActive, etc.)
+### Overview of Reanimated 3.x
 
-### Known problems and limitations
+The developments of Reanimated 3 were focused on improving the stability and performance. This version uses the Reanimated v2 API you know and love continuously embracing worklets and shared value architecture.
```
Comment (tomekzaw) [inline docs/docs/fundamentals/about.md:38]: Actually, initially the only reason to release v3 was that we didn't want to implement Fabric support for v1 API and we wanted to get rid of the v1 code. After some time, v2 and v3 branches diverged a bit. When we rewrote layout animations and shared values, we didn't want to waste our time on backporting the changes. So effectively v3 comes with stability and performance improvements as well as Fabric support.

**Review by tomekzaw — APPROVED** [review]

**Review by tomekzaw — CHANGES_REQUESTED** [review]
plz fix links (CI)

File: docs/docs/fundamentals/about.md (hunk 2 above)
Comment (kacperkapusciak) [reply docs/docs/fundamentals/about.md:38]: The effort you guys put into developments of Reanimated 3 resulted in stability and performance improvements. It sounds weird to emphasise "that was just a side effect".

Do you want me to rephrase the sentences to better represent 'the truth' or leave it as it is?

File: docs/docs/fundamentals/about.md (hunk 1 above)
Comment (kacperkapusciak) [reply docs/docs/fundamentals/about.md]: Fixed in [175b95f](https://github.com/software-mansion/react-native-reanimated/pull/4102/commits/175b95faacec0747737d821c760e9b087cbeee8d)

File: docs/docs/fundamentals/about.md (hunk 2 above)
Comment (tomekzaw) [reply docs/docs/fundamentals/about.md:38]: Nah, it's okay as it is. Just wanted to tell you the whole story.

File: docs/docs/fundamentals/about.md [hunk 3]
```diff
@@ -6,15 +6,15 @@ sidebar_label: About
...
 
-If you are looking for Reanimated 1 docs [please follow this link](https://docs.swmansion.com/react-native-reanimated/docs/1.x/).
+Looking for an older version? Check the [Reanimated 2.x](/docs/2.x/) or [Reanimated 1.x](/docs/1.x/).
 
 :::
 
 :::caution
 
-Reanimated v2 only supports `react-native` `0.62+`
+Reanimated **3.0.0** supports `react-native` `0.64+`. Subsequent releases aim to support [3 last stable React Native versions](https://github.com/reactwg/react-native-releases#which-versions-are-currently-supported).
```
Comment (tomekzaw) [inline docs/docs/fundamentals/about.md:17]:
```suggestion
Reanimated **3.0.0** supports React Native 0.64+. Subsequent releases aim to support [3 last stable React Native versions](https://github.com/reactwg/react-native-releases#which-versions-are-currently-supported).
```

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**